docs/architecture/pairing-with-use-integration-system.md

```diff
@@ -0,0 +1,86 @@
+
+## Solution overview
+
+This diagram illustrates the flow for pairing the Application with its Integration System Instance using the Pairing Adapter.
+
+![Integration System Instance Pairing Adapter](./assets/external-broker-pairing-adapter.svg)
+
+>**NOTE** The Application must be created before the pairing is performed. To create an Application that is managed by the Integration System the Application must contain the `integrationSystemID` attribute set to the corresponding value. It can be simplified by creating the Application from the appropriate ApplicationTemplate. 
+
+### Steps
+
+1. The Administrator requests the pairing data for the Application using the Cockpit (1).
+
+1. The Cockpit contacts the Director component to obtain the pairing data for the Application (2). The Application is loosely coupled with the Integration System that manages it. Thanks to this relation the Director component is aware which Pairing Adapter is assigned to the Integration System. Knowing this, the Pairing Adapter is called to request the pairing data (3).
+
+1. The Pairing Adapter calls Integration System Instance to obtain pairing data (4).
+
+1. Administrator after obtaining the pairing data provides them to the Application instance (5) which allows the Application instance to establish trusted relation with Integration System Instance (6). This informs Integration System Instance that there is a new Application instance that needs to be managed via this Integration System Instance.
+
+1. After successful pairing Integration System Instance can manage the Application (7).
+
+>**NOTE** The pairing process described here allows the Application to establish trusted relation with its Integration System Instance and does not establish trusted relation between the Application and its Runtimes.
+
+## Details
+
+The Compass supports multiple Integration System Instances (represented internally by Integration Systems) to help abstract the way how Applications and Runtimes can be managed. Each Integration System can have its own approach for obtaining details for establishing a trusted relation for the managed system (Application, Runtime). To make the Director architecture clean and improve users' experience we introduce the concept of Pairing Adapters. Each Integration System can be coupled with Pairing Adapter. Thanks to this approach the Directors' API can leverage different Integration Systems under single mutation and handle the requests in a consistent manner. When Administrator wants to establish a new trusted relation between Application and Integration System the Director API offers `generateAuthForApplication(id: ID!, type: PairingAdapterAuthTypeInput!, inputParams: JSON!)` mutation.
```

Comment:
`generateAuthForApplication` prototype is not consistent across all the documents within this PR. Please unify all occurrences.